www/app/app.js

```diff
@@ -562,6 +566,7 @@ define(['angularAMD', 'app.routes', 'app.constants', 'app.notifications', 'app.p
 			}
 		});
 		permissions.setPermissions(permissionList);
+		$rootScope.config.userName = permissionList.user;
```

Comment:
bit mixed feelings about this one.
You can set the userName within the function 'setPermissions', which I believe you already do now with the patch you made in app.permissions.js

Reply:
I had the same... but the 'permissions' object is never added to the rootscope, so not accessible in the html template
Therefor this was my way to get access to the Username, but open for better alternatives (here my lack of front-end/angular knowledge plays a part as well).

Reply:
Okey maybe as alternative, where you call 'getconfig' (app.js/$rootScope.GetGlobalConfig), also return the username, and store it in the rootScope config ?

Reply:
in the current order of processing (in the `app.run()`-callback method) that does not work as the order now is

- getConfig (json.htm?type=command&param=getconfig)
- getVersion (json.htm?type=command&param=getversion)
- getAuth (json.htm?type=command&param=getauth)
Only after the `getAuth` the Auth information is retrieved (incl. the Username) and stored into the Permission object.
And the getConfig does not retrieve this information so cannot set it into the rootScope config at that time.

Maybe we could consider to extend the 'getConfig' (which is _only_ used in the `app.js` controller) with information like the Version and Userinfo so everything can be retrieved in a single call (instead of 3 XHR calls at the moment). Would that be a better direction? (and we leave the other 2 as they are used for version checking and login/logout)

Reply:
hm... just looking at the getConfig command, it already uses the session information to get the correct Tabs for the specific User :) So it already has User specific processing which makes adding the Username easy.

Reply:
Yep that was my idea as well (to extend the returned info)

Reply:
done :) .. getConfig now includes the Username and all info from getVersion (simply by calling the command) and the UI no longer needs to do a separate request to getVersion when starting (saves a roundtrip)